apps/dashboard/config/application.rb

```diff
@@ -46,5 +46,17 @@ class Application < Rails::Application
       config.autoload_paths << ::Configuration.config_root.join("lib").to_s
       config.paths["app/views"].unshift ::Configuration.config_root.join("views").to_s
     end
+
+    # Enable installed plugins only if configured by administrator
+    plugins_dir = Pathname.new(::Configuration.plugins_directory)
+    if plugins_dir.directory?
+      plugins_dir.children.select(&:directory?).each do |installed_plugin|
+        next unless installed_plugin.readable?
+
+        config.paths["config/initializers"] << installed_plugin.join("initializers").to_s
+        config.autoload_paths << installed_plugin.join("lib").to_s
+        config.paths["app/views"].unshift installed_plugin.join("views").to_s
```

Comment:
I think we should likely have additional checks here for root owned files in production. I just want to be extra doubly sure that what we load in production is indeed given by an administrator.

```ruby
        init_dir = installed_plugin.join("initializers")
        safe_load = init_dir.children.all? { |f| File.stat(f.to_s).uid.zero? }
        config.paths["config/initializers"] << init_dir if safe_load
```

@treydock any additional thoughts here on loading trusted files?

Reply:
I understand the need to make sure that the configuration is given by an admin, but why is this different fromthe ones we use for `load_external_config?`, `::Configuration.config_root` (/etc/ood/config/apps/dashboard)?

Reply:
We should patch those too, we just haven't.

Reply:
I made a change to verify that the plugin folders are owned by root when in production